src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_button_press_notification.cc

```diff
@@ -52,155 +53,35 @@ OnButtonPressNotification::OnButtonPressNotification(
     rpc_service::RPCService& rpc_service,
     HMICapabilities& hmi_capabilities,
     policy::PolicyHandlerInterface& policy_handler)
-    : CommandNotificationImpl(message,
-                              application_manager,
-                              rpc_service,
-                              hmi_capabilities,
-                              policy_handler) {}
+    : ButtonNotificationToMobile(message,
+                                 application_manager,
+                                 rpc_service,
+                                 hmi_capabilities,
+                                 policy_handler) {}
 
 OnButtonPressNotification::~OnButtonPressNotification() {}
 
-void OnButtonPressNotification::Run() {
+void OnButtonPressNotification::SendButtonNotification(
+    ApplicationSharedPtr app) {
   SDL_LOG_AUTO_TRACE();
 
-  const uint32_t btn_id = static_cast<uint32_t>(
-      (*message_)[strings::msg_params][hmi_response::button_name].asInt());
+  mobile_apis::ButtonPressMode::eType btn_press_mode;
```

Comment:
I believe you can move all this code to a base class as a new function, which accepts msg_params and function_id as an input and returns smart object which should be sent

Reply:
If idea is unify methods SendButtonNotification() from classes OnButtonEventNotification and OnButtonPressNotification then we faced to some diffs in implementations:

in OnButtonEventNotification we retrieve button_event_mode to 
mobile_apis::ButtonEventMode::eType btn_event_mode;

in OnButtonPressNotification we retrieve button_press_mode to 
mobile_apis::ButtonPressMode::eType btn_press_mode;

Seems there are not many common lines of code